libbpf-rs/src/object.rs

```diff
@@ -283,6 +283,25 @@ pub struct Object {
     progs: HashMap<String, Program>,
 }
 
+fn object_prog_next(
+    obj: *const libbpf_sys::bpf_object,
+    prog: *mut libbpf_sys::bpf_program, // needed to know what the next program is
+) -> Result<Option<*mut libbpf_sys::bpf_program>> {
+    let next_prog_ptr: *mut libbpf_sys::bpf_program = unsafe { libbpf_sys::bpf_program__next(prog, obj) };
+    Ok(util::ptr_to_option(next_prog_ptr))
+}
+
+fn object_map_next(
+    obj: *const libbpf_sys::bpf_object,
+    map: *mut libbpf_sys::bpf_map, // needed to know what the next map is
+) -> Result<Option<*mut libbpf_sys::bpf_map>> {
+    let next_map_ptr: *mut libbpf_sys::bpf_map = unsafe { libbpf_sys::bpf_map__next(map, obj) };
+    Ok(util::ptr_to_option(next_map_ptr))
+}
```

Comment:
I don't think we need helpers for this. First, this always returns `Ok(..)` so the `Result<..>` doesn't do anything. And second, we can just do:
```
loop {
  ...
  let next_map_ptr: ... = unsafe { libbpf_sys::bpf_map__next(...) };
  if next_map_ptr.is_null() {
    break;
  }
  ...
}
```
in the code below